java-checks-test-sources/default/src/main/java/checks/spring/StatusCodesOnResponseCheckSample.java

```diff
@@ -24,6 +24,52 @@ public ResponseEntity<User> getBadReqUser() {
       return ResponseEntity.badRequest().build(); // Compliant
     }
 
+    public ResponseEntity<User> foo() {
+      User user = getUserObject();
+      if (user == null) {
+        return ResponseEntity.notFound().build(); // Compliant
```

Comment:
I think we should add a comment here to explain that we cannot raise on invocation inside of if statements alone

Reply:
Let's describe in the comments why we do not raise on invocations in `if` statements